clinica/pydra/t1_volume/t1_volume_utils.py

```diff
@@ -0,0 +1,536 @@
+        parameters["save_warped_unmodulated"],
+        parameters["save_warped_modulated"],
+    )
+    return tissue_tuples
+
+
+def init_input_node(bids_name: str) -> None:
+    """Extract "sub-<participant_id>_ses-<session_label>" from <bids_name> and print begin message.
+
+    Parameters
+    ----------
+    bids_name : str
+        The BIDS name of the t1w image
+
+    Warns
+    -----
+    Outputs ID of subjet/session being processed
+    """
+
+    from clinica.utils.filemanip import get_subject_id
+    from clinica.utils.ux import print_begin_image
+
+    subject_id = get_subject_id(bids_name)
+    print_begin_image(subject_id)
+
+    return subject_id, bids_name
+
+
+def print_end_pipeline(bids_name: str) -> None:
+    """Prints end message.
+
+    Parameters
+    ----------
+    bids_name : str
+        The BIDS name of the t1w image
+
+    Warns
+    -----
+    Prints out message for image <bids_name>
+    """
+    from clinica.utils.ux import print_end_image
+
+    print_end_image(bids_name)
+
+
+def zip_list_files(class_images: list, zip_files: bool = False) -> list:
+    """Create list of (Optionally) zipped nifti images
+    Parameters
+    ----------
+    class_images : list
+    zip_files: bool
+
+    Returns
+    -------
+    list
+        list of (optionally) zipped nifti files
+    """
+    from clinica.utils.filemanip import zip_nii
+
+    if zip_files:
+        return [zip_nii(tissue, True) for tissue in class_images]
+
+    return [tissue for tissue in class_images]
+
+
+def get_tissue_tuples(
+    tissue_map: PathLike,
+    tissue_classes: list,
+    dartel_tissues: list,
+    save_warped_unmodulated: bool,
+    save_warped_modulated: bool,
+) -> list:
+    """Extract list of tuples, one for each tissue clas.
+
+    Parameters
+    ---------
+    tissue_map : PathLike
+        Path to tissue maps
+    tissue_classes: list
+        Classes of images to obtain from segmentation. Ex: [1,2,3] is GM, WM and CSF
+    dartel_tissues: list
+        Classes of images to save for DARTEL template calculation. Ex: [1] is only GM'
+    save_warped_unmodulated : bool
+        Save warped unmodulated images for tissues specified in --tissue_classes
+    save_warped_modulated: bool
+        Save warped modulated images for tissues specified in --tissue_classes
+
+    Returns
+    -------
+    list
+        List of tuples according to NewSegment input por tissues
+
+    Notes
+    -----
+     The returned list contains tissue probability map (4D), 1-based index to frame
+     - number of gaussians
+     - which maps to save [Native, DARTEL] - a tuple of two boolean values
+     - which maps to save [Unmodulated, Modulated] - a tuple of two boolean values
+
+    """
+    tissues = []
+
+    for i in range(1, 7):
+        n_gaussians = 2
+
+        if i == 4 or i == 5:
+            n_gaussians = i - 1
+
+        native_space = False
+        dartel_input = False
+        warped_unmodulated = False
+        warped_modulated = False
```

Comment:
```suggestion
        warped_modulated = (i in tissue_classes) and save_warped_modulated
```

And delete the two if blocks afterwards.